diff --git a/scripts/bootstrap/termux-bootstrap-second-stage.sh b/scripts/bootstrap/termux-bootstrap-second-stage.sh
index 723993d6dc..e0bec45b5e 100755
--- a/scripts/bootstrap/termux-bootstrap-second-stage.sh
+++ b/scripts/bootstrap/termux-bootstrap-second-stage.sh
@@ -149,10 +149,65 @@ run_bootstrap_second_stage_inner() {
 		return $return_value
 	fi
 
+	log "Restoring ct.sym"
+	restore_ctsyms
+	return_value=$?
+	if [ $return_value -ne 0 ]; then
+		log_error "Failed to restore ct.sym"
+		return $return_value
+	fi
+
+	log "Repacking archives"
+	repack_archives
+	return_value=$?
+	if [ $return_value -ne 0 ]; then
+		log_error "Failed to repack archives"
+		return $return_value
+	fi
+
 	return 0
 
 }
 
+repack_archives() {
+	# Find and repack all ZIP files with max compression
+	while read -r -d '' zip; do
+		repack_zip "$(realpath "$zip")"
+	done < <(find "$TERMUX_PREFIX" -type f -name \*.zip -print0)
+}
+
+repack_zip() {
+	input="$1"
+
+	log "Repacking $input"
+
+	local temp
+	temp=$(mktemp -d)
+
+	(cd "$temp"
+		unzip -qq "$input"
+		rm "$input"
+		zip -XDqr9 "$input" ./*
+	)
+}
+
+restore_ctsyms() {
+	# Repack ct.sym in all installed JDKs
+	for ctsym in "$TERMUX_PREFIX"/lib/jvm/java-*-openjdk/lib/ct.sym; do
+		[[ -f "$ctsym" ]] || continue
+		ctsym="$(realpath "$ctsym")"
+		local temp
+		temp=$(mktemp -d)
+		(cd "$temp"
+			brotli -do ct.sym "$ctsym"
+			unzip -qq "ct.sym"
+			rm "ct.sym"
+			rm "$ctsym"
+			zip -qr9 "$ctsym" ./*
+		) && rm -rf "$temp" && log "Restored ct.sym in $(basename "$(dirname "$(dirname "$ctsym")")")"
+	done
+}
+
 run_package_postinst_maintainer_scripts() {
 
 	local return_value
